receiver/mysqlreceiver/client.go

```diff
@@ -38,11 +38,12 @@ var _ client = (*mySQLClient)(nil)
 
 func newMySQLClient(conf *Config) client {
 	driverConf := mysql.Config{
-		User:   conf.Username,
-		Passwd: conf.Password,
-		Net:    conf.Transport,
-		Addr:   conf.Endpoint,
-		DBName: conf.Database,
+		User:                 conf.Username,
+		Passwd:               conf.Password,
+		Net:                  conf.Transport,
+		Addr:                 conf.Endpoint,
+		DBName:               conf.Database,
+		AllowNativePasswords: true,
```

Comment:
It seems wrong to just add this without giving users the chance to customize it. Besides, the default value for this seems to indeed be true: what would this change bring?

https://pkg.go.dev/github.com/go-sql-driver/mysql#Config

Reply:
It seems that the default was incorrect in testing as it was building the DSN by setting this parameter to false. I'll add it as a config option but default to true.